tests/rptest/tests/double_bytes_sent_test.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random
+from ducktape.mark.resource import cluster
+from ducktape.utils.util import wait_until
+
+from rptest.services.rpk_producer import RpkProducer
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.types import TopicSpec
+
+
+class BytesSentTest(RedpandaTest):
+    """
+    Produce and consume some data then confirm the bytes sent metric.
+    """
+    topics = (TopicSpec(replication_factor=1), )
+
+    def __init__(self, test_context):
+        super(BytesSentTest, self).__init__(test_context=test_context,
+                                            num_brokers=1)
+
+        self._producer = RpkProducer(test_context,
+                                     self.redpanda,
+                                     self.topic,
+                                     msg_size=16384,
+                                     msg_count=1)
+
+        # Maxes were calculated by running the BytesSentTest locally
+        # and observing how many bytes were sent after
+        # startup (first max) and after producing 10 records (second max)
+        self._first_max = 1984
+        self._second_max = 2519
+
+    def _bytes_sent(self):
+        # Convert the metrics generator to a list
+        metrics = list(self.redpanda.metrics(self.redpanda.nodes[0]))
+
+        # Find the metric family that tracks bytes sent from kafka subsystem
+        family_filter = filter(
+            lambda fam: fam.name == "vectorized_kafka_rpc_sent_bytes", metrics)
+        family = next(family_filter)
+
+        # Sum bytes sent
+        bytes_sent = 0
+        for sample in family.samples:
+            bytes_sent += sample.value
+
+        return bytes_sent
+
+    @cluster(num_nodes=2)
+    def test_bytes_sent(self):
+
+        # Startup checkpoint for bytes sent
+        self.logger.debug(f"First bytes_sent: {self._bytes_sent()}")
+        assert self._bytes_sent() <= self._first_max
+
+        # Produce some data
+        self._producer.start()
+        self._producer.wait()
+
+        # Producer checkpoint for bytes sent
+        self.logger.debug(f"Second bytes_sent: {self._bytes_sent()}")
+        assert self._bytes_sent() <= self._second_max
```

Comment:
I'd suggest changing the strategy slightly to make the test more robust and generic. i think it will also allow you to remove the empirically derived thresholds.

```python
# write roughly 5 mb of 512 byte recores
self.kafka_tools.produce(self.topic, 10240, 512, acks=-1)

# establish the _current_ counter which won't be zero, but
# we don't really care what the actual value is
start_bytes = self._bytes_sent()

# consume some data with a rough estimation of what the size
# would be. for example, 2048 * 512 records = 1 MB
bytes = consume(self.topic, 2048)
bytes_received = count_actual_bytes_received(bytes)

end_bytes = self._bytes_sent()
total_sent = end_bytes - start_bytes

assert total_sent > bytes_received
assert total_sent is within a small % threshold of bytes_received
```